soh/soh/Enhancements/randomizer/randomizer.cpp

```diff
@@ -440,6 +441,116 @@ void Randomizer::LoadMerchantMessages(const char* spoilerFileName) {
             "%gobjet mystérieux%w pour 60 Rubis?\x1B&%gOui&Non%w",
         });
 
+    //Setup for merchant text boxes
+    //Medigoron
+    //RANDOTODO: Implement obscure/ambiguous hints
+    RandomizerGet medigoronGet = this->itemLocations[RC_GC_MEDIGORON].rgID;
+    std::vector<std::string> mgItemName;
+    if (Randomizer::GetRandoSettingValue(RSK_SHUFFLE_MERCHANTS) == 2) {//If merchant hints are clear... 
+        if (medigoronGet == RG_ICE_TRAP) {
+            medigoronGet = this->itemLocations[RC_GC_MEDIGORON].fakeRgID;
+            mgItemName = {
+                this->itemLocations[RC_GC_MEDIGORON].trickName, 
+                this->itemLocations[RC_GC_MEDIGORON].trickName,
+                this->itemLocations[RC_GC_MEDIGORON].trickName
+            };
+        } else {
+            mgItemName = EnumToSpoilerfileGetName[medigoronGet];
+        }
+    } else {
+        mgItemName = {
+            "something cool",
+            "etwas cooles",
+            "un truc cool"
+        };
+    }
+         
+    CustomMessageManager::Instance->CreateMessage(
+        Randomizer::merchantMessageTableID, TEXT_MEDIGORON,
+        {
+            TEXTBOX_TYPE_BLACK,
+            TEXTBOX_POS_BOTTOM,
+            "How about buying %r&" + mgItemName[0] + "%w for %g200 rupees%w?\x1B&%gYes&No%w",
```

Comment:
Lets adapt this to use the new `ReplaceStringInMessage` pattern introduced in #1847, and have the item name filled in at retrieval rather than on the message creation.